### PR TITLE
Fix mismatch fit3, add mismatch fit2

### PR DESCRIPTION
**Pull request by echedey-ls**
From discussion at https://github.com/pvlib/pvlib-python/pull/2147

Fixes results from the `bifacial_radiance.mismatch.mismatch_fit3`, specifies output from this function to be in percentage, per paper.

Adds `.mismatch_fit2`, which is recommended for all scenarios, per paper.

Test strategy is just taking the output from the functions after the alleged fixes and putting it into the expected values, and fit3 is compared to the code implemented in the latest changes made to pvlib in PR linked on top. Fit 2 is a copy of fit 3 with changed coefficients.

### CODE REVIEW

**Comment by echedey-ls** [comment]
@cdeline @shirubana @kandersolar can any of you approve the workflows and evaluate if these changes are of any interest to the project?

**Comment by cdeline** [comment]
Tests are failing because of issues in the Main branch, but this has been fixed with pull #522.  Please merge from Main and I can approve this.  Thanks!

File: bifacial_radiance/mismatch.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def mismatch_fit3(data):
     Returns
     -------
     fit3 :  Float or pd.Series
-        Returns mismatch values for each module
+        Returns mismatch values for each module in percentage [%].
     
     Equation: 1/(n^2*Gavg)*Sum Sum (abs(G_i - G_j))
     ## Note: starting with Pandas 1.0.0 this function will not work on Series objects.
```
Comment (cdeline) [inline bifacial_radiance/mismatch.py:185]: Please include this deprecation message:
    ## Note: This fit value is deprecated in favor of Fit2 which has better agreement
    ##       with the experimental data.

Comment (echedey-ls) [reply bifacial_radiance/mismatch.py:189]: There it is with a Sphinx directive.

I haven't seen any deprecation module so deprecation tests haven't been added.

**Comment by echedey-ls** [comment]
Feel free to make all the changes you want to the whatsnew.

**Comment by cdeline** [comment]
Pytests are passing aside from the Docker test, which won't work correctly when pulling from a fork, per (https://github.com/docker/login-action/issues/29) so this is good to go.  Merging.

**Comment by cdeline** [comment]
Thanks @echedey-ls!  This will go into a bugfix release 0.4.4 as soon as we are able to. 🎉

**Pull request merged**